File: 排序算法/冒泡排序/Go/BubbleSort.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Scan()
	numsStrArr := strings.Split(scanner.Text(), " ")
	var numsIntArr []int64
	for _, value := range numsStrArr {
		v, _ := strconv.ParseInt(value, 10, 64)
		numsIntArr = append(numsIntArr, v)
	}
	numsIntArr = bubbleSort(numsIntArr)
	fmt.Println(numsIntArr)
}

func bubbleSort(arr []int64) []int64 {
	if arr == nil || len(arr) < 2 {
		return arr
	}
	length := len(arr)
	for i := 0; i < length-1; i++ {
		for j := 0; j < length-i-1; j++ {
			if arr[j] > arr[j+1] {
				temp := arr[j]
				arr[j] = arr[j+1]
				arr[j+1] = temp
			}
		}
	}
	return arr
}

File: 排序算法/冒泡排序/Java/src/BubbleSort.java
import java.util.Arrays;
import java.util.Scanner;

public class BubbleSort {
    public static void main(String[] args) {
        String[] strArr = new Scanner(System.in).nextLine().split(" ");
        int length = strArr.length;
        int[] intArr = new int[length];
        for (int i = 0; i < length; i++) {
            intArr[i] = Integer.parseInt(strArr[i]);
        }
        bubbleSort(intArr);
        System.out.println(Arrays.toString(intArr));
    }

    public static void bubbleSort(int[] arr) {
        if (arr == null || arr.length < 2) return;
        int length = arr.length;
        for (int i = 0; i < length - 1; i++) {
            for (int j = 0; j < length - 1 - i; j++) {
                if (arr[j] > arr[j + 1]) {  // 升序
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                }
            }
        }
    }
}
